doc/source/roadmap.rst

```diff
@@ -1 +1,104 @@
+Apple doesn't update Accelerate anymore). OpenBLAS isn't very stable though,
+sometimes its releases break things and it has issues with threading (currently
+the only issue for using SciPy with PyPy3).  We need at the very least better
+support for debugging OpenBLAS issues, and better documentation on how to build
+SciPy with it.  And option is to use BLIS for a BLAS interface (see `numpy
+gh-7273 <https://github.com/numpy/numpy/issues/7372>`__).
+
+2. Support for newer LAPACK features.  In SciPy 1.2.0 we increased the minimum
+supported version of LAPACK to 3.4.0.  Now that we dropped Python 2.7, we can
+increase that version further (MKL + Python 2.7 was the blocker for >3.4.0
+previously) and start adding support for new features in LAPACK.
+
+
+Sparse arrays
+-------------
+
+The sparse matrix formats are mostly feature-complete, however the main issue
+is that they act like ``numpy.matrix`` (which will be deprecated in NumPy at
+some point).  What we want is sparse *arrays* that act like ``numpy.ndarray``.
+This is being worked on in https://github.com/pydata/sparse, which is quite far
+along.  The tentative plan is:
+
+- Start depending on ``pydata/sparse`` once it's feature-complete enough (it
+  still needs a CSC/CSR equivalent) and okay performance-wise.
+- Add support for ``pydata/sparse`` to ``scipy.sparse.linalg`` (and perhaps to
+  ``scipy.sparse.csgraph`` after that).
+- Indicate in the documentation that for new code users should prefer
+  ``pydata/sparse`` over sparse matrices.
+- When NumPy deprecates ``numpy.matrix``, vendor that or maintain it as a
+  stand-alone package.
+
+
+Fourier transforms
+------------------
+
+We want to integrate PocketFFT into ``scipy.fftpack`` for significant
+performance improvements (see `this NumPy PR
+<https://github.com/numpy/numpy/pull/11888>`__ for details),
+add a backend system to support PyFFTW and mkl-fft,
+and align the function signatures of ``numpy.fft`` and ``scipy.fftpack``.
+
+
+Support for distributed arrays and GPU arrays
+---------------------------------------------
+
+NumPy is splitting its API from its execution engine with
+``__array_function`` and ``__array_ufunc__``.  This will enable parts of SciPy
+to accept distributed arrays (e.g. ``dask.array.Array``) and GPU arrays (e.g.
+``cupy.ndarray``) that implement the ``ndarray`` interface.  At the moment it is
+not yet clear which algorithms will work out of the box, and if there are
+significant performance gains when they do.  We want to create a map of which
+parts of the SciPy API work, and improve support over time.
+
+
+Improve benchmark system for ``optimize``
+-----------------------------------------
+
+``scipy.optimize`` has an extensive set of benchmarks for accuracy and speed of
+the global optimizers. That has allowed adding new optimizers (``shgo`` and
+``dual_annealing``) with significantly better performance than the existing
+ones.  The benchmark system itself is slow and hard to use however; we need to
```

Comment:
`The benchmark system itself` here means `asv` or something else?

Reply:
Specifically the `optimize` benchmarks. You need to set `SCIPY_XSLOW`, use `SCIPY_GLOBAL_BENCH`, and then it takes forever. Plus there's no good way to compare outputs. For `shgo` and `dual_annealing` both authors forked the benchmarks so they could actually work with them.

I'll add `optimize` to "the benchmark system" to clarify.

Reply:
I'm actually not 100% sure that this global benchmarks idea needs to be here - probably the least important of everything I put in this list.

Reply:
I may be biased but I wouldn't say that the shgo and dual_annealing are *significantly* better than basinhopping or differential_evolution. THey're all pretty good.

Reply:
Yeah, the global benchmarks are not straightforward to use, could definitely do with more documentation.

Reply:
It would be great if we could actually get #8779 working. It could serve as a way to show people how to run the benchmarks.

Reply:
> It would be great if we could actually get #8779 working. It could serve as a way to show people how to run the benchmarks.

+1 I would love to know how to run the benchmarks

Reply:
> I may be biased but I wouldn't say that the shgo and dual_annealing are significantly better than basinhopping or differential_evolution. 

Based on the recent benchmarks, `basinhopping` < `differential_evolution` < `dual_annealing` / `shgo`. After using the new additions recently, I'm not yet convinced that `dual_annealing` is significantly better than `differential_evolution`, but I'm quite impression by `shgo` (also its ability to find *all* minima is very handy.

But yeah, I remember the pre-`basinhopping` days, compared to back then they're all awesome:)

Reply:
> +1 I would love to know how to run the benchmarks

The basics are straightforward: `python runtests.py --bench`. However, when we get into comparing benchmarks between branches etc., it tends to get fragile.